lib/ethpy/ethpy/hyperdrive/api.py

```diff
@@ -454,15 +468,22 @@ def bonds_given_shares_and_rate(self, target_rate: FixedPoint) -> FixedPoint:
         ---------
         target_rate : FixedPoint
             The target apr for which to calculate the bond reserves given the pools current share reserves.
+        target_shares : FixedPoint, optional
+            The target share reserves for the pool
         """
-
-        mu: FixedPoint = self.pool_config["initialSharePrice"]
-        z_minus_zeta: FixedPoint = self.pool_info["shareReserves"] - self.pool_info["shareAdjustment"]
-        t = self.position_duration_in_years
-        one_over_tau: FixedPoint = self.pool_config["timeStretch"]
-        adjusted_apr = FixedPoint("1") + target_rate * t
-
-        return mu * z_minus_zeta * adjusted_apr**one_over_tau
+        if target_shares is None:
+            target_shares = self.effective_share_reserves
+        return FixedPoint(
+            scaled_value=int(
+                pyperdrive.calculate_bonds_given_shares_and_rate(  # pylint: disable=no-member
+                    str(target_shares.scaled_value),
```

Comment:
Is the int -> str mapping required for pyperdrive functions? Is there a way we can handle this under the hood in pyperdrive?

Reply:
yes, discussed here: https://github.com/delvtech/elf-simulations/issues/905